File: HotelBookingSystem/src/Shared/HotelBooking.Models/Entities/Booking.cs
using System;
using System.Collections.Generic;

namespace HotelBooking.Models.Entities
{
    public class Booking
    {
        public Guid BookingId { get; set; } = Guid.NewGuid();
        public Guid CustomerId { get; set; }
        public Guid RoomId { get; set; }
        public DateTime CheckInDate { get; set; }
        public DateTime CheckOutDate { get; set; }
        public int NumberOfGuests { get; set; }
        public decimal TotalAmount { get; set; }
        public string BookingStatus { get; set; } = "Confirmed"; // Confirmed, Cancelled, CheckedIn, CheckedOut
        public bool IsRecurring { get; set; } = false;
        public string? RecurrencePattern { get; set; } // Weekly, Monthly, etc.
        public string BookingReference { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;

        // Navigation properties
        public virtual Customer Customer { get; set; } = null!;
        public virtual Room Room { get; set; } = null!;
        public virtual ICollection<SpecialRequest> SpecialRequests { get; set; } = new List<SpecialRequest>();
    }
}

File: HotelBookingSystem/hotel-booking-frontend/src/types/booking.ts
export interface Booking {
  bookingId: string;
  customerId: string;
  roomId: string;
  checkInDate: string;
  checkOutDate: string;
  numberOfGuests: number;
  totalAmount: number;
  bookingStatus: 'Confirmed' | 'Cancelled' | 'Pending' | 'CheckedIn' | 'CheckedOut';
  isRecurring: boolean;
  recurrencePattern?: string;
  bookingReference: string;
  createdAt: string;
  updatedAt: string;
  customer?: Customer;
  room?: Room;
  specialRequests?: SpecialRequest[];
}

export interface CreateBookingDto {
  customerId: string;
  roomId: string;
  checkInDate: string;
  checkOutDate: string;
  numberOfGuests: number;
  isRecurring?: boolean;
  recurrencePattern?: string;
}

export interface SpecialRequest {
  requestId: string;
  bookingId: string;
  requestType: string;
  description: string;
  status: 'Pending' | 'Approved' | 'Fulfilled' | 'Rejected';
  requestDate: string;
  fulfilledDate?: string;
}

export interface CreateSpecialRequestDto {
  requestType: string;
  description: string;
}

import { Customer } from './customer';
import { Room } from './room';
